src/Components/Header/Menu/index.jsx

```diff
@@ -7,55 +7,48 @@ import DifficultyMenu from './Difficulty';
 import TagMenu from './Tags';
 import Options from './OptionsMenu';
 import ProfileMenu from './ProfileMenu';
+import SearchBar from './Search';
 
-const style = {
-  disabled_tag: {
-    opacity: '0.2',
-  },
-  menu_item: {
-    cursor: 'default',
-  },
-  empty_space: {
-    width: '4.5vw',
-  },
+const TagChoice = ({ loggedIn }) => {
+  if (loggedIn) { return <TagMenu />; }
+  return (
+    <Popup
+      trigger={(
+        <MenuItem>
+          <Icon
+            name="tags"
+            style={{ opacity: '0.2' }}
+            size="big"
+          />
+        </MenuItem>
+)}
+      inverted
+      content="Sign up to save poses to your favorites"
+    />
+  );
 };
 
-const MenuItem = styled(Menu.Item)`
-  cursor: default;
-`;
-
-const HeaderMenu = ({ name, difficulty }) => (
+const HeaderMenu = ({ userName, difficulty }) => (
   <Fragment>
     <DifficultyMenu difficultySetting={difficulty} />
-    {name ? (
-      <TagMenu />
-    ) : (
-      <Popup
-        trigger={(
-          <MenuItem>
-            <Icon
-              name="tags"
-              style={style.disabled_tag}
-              size="big"
-            />
-          </MenuItem>
-)}
-        inverted
-        content="Sign up to save poses to your favorites"
-      />
-    )}
+    <TagChoice loggedIn={userName} />
+    <SearchBar />
     <Menu.Menu position="right">
-      {!name && <ProfileMenu />}
-      <Options isUser={name} />
+      {!userName && <ProfileMenu />}
+      <Options isUser={userName} />
     </Menu.Menu>
   </Fragment>
 );
 
-const mapStateToProps = ({ view: { difficulty }, user: { name } }) => ({ difficulty, name });
+const mapStateToProps = ({ view: { difficulty }, user: { name: userName } }) => ({ difficulty, userName });
 const mapDispatchToProps = dispatch => ({
   UserLogout: () => dispatch({
     type: LOG_OUT,
   }),
 });
 
 export default connect(mapStateToProps, mapDispatchToProps)(HeaderMenu);
+
+const MenuItem = styled(Menu.Item)`
```

Comment:
styles on top

Reply:
I don't like styles on top since they're not the first thing I want to see on a component normally. I'll just move them to a separate file